pkg/plugin/resources.go

```diff
@@ -341,73 +400,175 @@ func (app *App) handleSaveLLMState(w http.ResponseWriter, req *http.Request) {
 		b, err := io.ReadAll(resp.Body)
 		if err != nil {
 			handleError(w, fmt.Errorf("failed to read response body to bytes %w", err), http.StatusInternalServerError)
 			return
 		}
-		handleError(w, fmt.Errorf("failed to save state in gcom: %s %s", resp.Status, string(b)), http.StatusInternalServerError)
+		handleError(w, fmt.Errorf("failed to save state in llm-gateway: %s %s", resp.Status, string(b)), http.StatusInternalServerError)
 		return
 	}
-	log.DefaultLogger.Debug("Saved state in gcom", "status", resp.Status)
+	log.DefaultLogger.Debug("Saved state in llm-gateway", "status", resp.Status)
 
 	// write a success response body since backendSrv.* needs a valid json response body
 	w.WriteHeader(http.StatusOK)
-	_, err = w.Write([]byte(`{"status": "Success"}`))
+	_, _ = w.Write([]byte(`{"status": "Success"}`))
 	if err != nil {
 		handleError(w, fmt.Errorf("failed to write response body %w", err), http.StatusInternalServerError)
 		return
 	}
 }
 
+func (a *App) handleLLMState(w http.ResponseWriter, req *http.Request) {
+	switch req.Method {
+	case "GET":
+		a.handleGetLLMOptInState(w, req)
+	case "POST":
+		a.handleSaveLLMOptInState(w, req)
+	default:
+		handleError(w, fmt.Errorf("method not allowed: %s", req.Method), http.StatusMethodNotAllowed)
+		return
+	}
+}
+
+// provisionedPlugin is the response returned by a call to grafana.com's provisioned-plugin's endpoint
+// for an specific instance's plugin.
+type provisionedPlugin struct {
+	ID int `json:"id"`
+}
+
+// getPluginID gets the *ID* of the *provisioned plugin* from grafana.com.
+// Note that this differs to the plugin ID referred to by the `backend.CallResourceRequest`,
+// which is 'grafana-ml-app'
+func getPluginID(ctx context.Context, slug string, grafanaAppURL string, saToken string, gcomAPIKey string) (int, error) {
+	gcomPath := "/api/gnet/instances/" + slug + "/provisioned-plugins/grafana-llm-app"
+	req, err := http.NewRequestWithContext(ctx, "GET", grafanaAppURL+gcomPath, nil)
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", saToken))
+	req.Header.Set("X-Api-Key", gcomAPIKey)
+	if err != nil {
+		return 0, fmt.Errorf("create http request: %w", err)
+	}
+	respBody, err := doRequest(req)
+	if err != nil {
+		return 0, err
+	}
+	plugin := provisionedPlugin{}
+	err = json.Unmarshal(respBody, &plugin)
+	if err != nil {
+		return 0, fmt.Errorf("unmarshal json: %w", err)
+	}
+	return plugin.ID, nil
+}
+
+func (a *App) handleSavePluginSettings(w http.ResponseWriter, req *http.Request) {
+	if req.Method != http.MethodPost {
+		handleError(w, fmt.Errorf("method not allowed: %s", req.Method), http.StatusMethodNotAllowed)
+		return
+	}
+
+	// Read the request body
+	if req.Body == nil {
+		log.DefaultLogger.Warn("Request body is nil")
+		handleError(w, errors.New("request body required"), http.StatusBadRequest)
+		return
+	}
+
+	log.DefaultLogger.Debug("Getting provisioned plugin ID from grafana.com")
+
+	notHG := os.Getenv("NOT_HG")
```

Comment:
We shouldn't require OSS users to set `NOT_HG` to make this work, and should instead either use our existing flag in the settings, or look for something like `HG_INSTANCE_SLUG` as the environment variable. If choosing the latter check with the Hosted Grafana team to identify the best environment variable to use.

Reply:
I added the NOT_HG to attempt to make local dev work.

Checked a dev hg instance, it should be `HG_INSTANCE_SLUG`, will use this instead.